prime-router/src/test/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctionsTests.kt

```diff
@@ -557,4 +560,94 @@ class CustomFHIRFunctionsTests {
             )
         }.isFailure().hasClass(SchemaException::class.java)
     }
+
+    @Test
+    fun `test convertDateToAge in future`() {
+        val currentDate = Date()
+        val calendar = Calendar.getInstance()
+        calendar.time = currentDate
+        calendar.add(Calendar.DATE, 1)
+
+        assertThat {
+            CustomFHIRFunctions.convertDateToAge(
+                mutableListOf(DateType(calendar.time))
+            )
+        }.isFailure().hasClass(SchemaException::class.java)
+    }
+
+    @Test
+    fun `test convertDateToAge year`() {
+        val currentDate = Date()
+        val calendar = Calendar.getInstance()
+        calendar.time = currentDate
+        calendar.add(Calendar.YEAR, -76)
+
+        val age = CustomFHIRFunctions.convertDateToAge(
+            mutableListOf(DateType(calendar.time))
+        )
+
+        assertThat(age is Age)
+        if (age is Age) {
```

Comment:
`age is Age` is returning false here. For the assert to work you'd have to change it to `assertThat(age is Age).isEqualTo(true)`. Once that starts working asserts inside the if are going to break, you have to change one of them to check for `age.code`.

Reply:
`if (age is Age)` also needs to be updated to `age[0] is Age`